src/NextjsDistribution.ts

```diff
@@ -0,0 +1,683 @@
+      // cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+      originRequestPolicy: lambdaOriginRequestPolicy,
+      compress: true,
+      // cachePolicy: lambdaCachePolicy,
+      cachePolicy: cloudfront.CachePolicy.CACHING_DISABLED,
+      edgeLambdas: lambdaOriginEdgeFns,
+    };
+
+    // requests to fallback origin group (default behavior)
+    // used for S3 and lambda. would prefer to forward all headers to lambda but need to strip out host
+    // TODO: try to do this with headers whitelist or edge lambda
+    const fallbackOriginRequestPolicy = new cloudfront.OriginRequestPolicy(this, 'FallbackOriginRequestPolicy', {
+      cookieBehavior: cloudfront.OriginRequestCookieBehavior.all(), // pretty much disables caching - maybe can be changed
+      queryStringBehavior: cloudfront.OriginRequestQueryStringBehavior.all(),
+      // we cannot forward the host header to a lambda URL
+      headerBehavior: cloudfront.OriginRequestHeaderBehavior.allowList(
+        'Accept',
+        'Referer',
+        'User-Agent',
+        'Content-Type'
+      ),
+      comment: 'Nextjs Fallback Origin Request Policy',
+    });
+
+    // if we don't have a static file called index.html then we should
+    // redirect to the lambda handler
+    const hasIndexHtml = this.props.nextBuild.readPublicFileList().includes('index.html');
+
+    return new cloudfront.Distribution(this, 'Distribution', {
+      // defaultRootObject: "index.html",
+      defaultRootObject: '',
+
+      // Override props.
+      ...cfDistributionProps,
+
+      // these values can NOT be overwritten by cfDistributionProps
+      domainNames,
+      certificate: this.certificate,
+      defaultBehavior: {
+        origin: fallbackOriginGroup, // try S3 first, then lambda
+        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        // allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL, // doesn't work with an OriginGroup
+        // cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+        compress: true,
+
+        // what goes here? static or lambda?
+        cachePolicy: lambdaCachePolicy,
+        originRequestPolicy: fallbackOriginRequestPolicy,
+
+        // edgeLambdas: lambdaOriginEdgeFns,
+      },
+
+      additionalBehaviors: {
+        // is index.html static or dynamic?
+        ...(hasIndexHtml ? {} : { '/': lambdaBehavior }),
+
+        // known dynamic routes
+        'api/*': lambdaBehavior,
+        '_next/data/*': lambdaBehavior,
+
+        // dynamic images go to lambda
+        '_next/image*': {
+          viewerProtocolPolicy,
+          origin: serverFunctionOrigin,
+          allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+          cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+          compress: true,
+          cachePolicy: imageCachePolicy,
+          originRequestPolicy: cloudfront.OriginRequestPolicy.ALL_VIEWER, // not sure what goes here
```

Comment:
From my experience, having ALL_VIEWER results in `{ "Message": null }` response. See the image optimization PR for the policy that works.